src/exporters/cds.py

```diff
@@ -0,0 +1,309 @@
+import cdsapi
+from pathlib import Path
+import certifi
+import urllib3
+import warnings
+import itertools
+from pprint import pprint
+
+from typing import Dict, Optional, List
+
+from .base import BaseExporter, Region
+
+http = urllib3.PoolManager(
+    cert_reqs='CERT_REQUIRED',
+    ca_certs=certifi.where()
+)
+
+
+class CDSExporter(BaseExporter):
+    """Exports for the Climate Data Store
+
+    cds.climate.copernicus.eu
+    """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+        super().__init__(data_folder)
+
+        self.client = cdsapi.Client()
+
+    @staticmethod
+    def get_era5_times(granularity: str = 'hourly') -> Dict:
+        """Returns the era5 selection request arguments
+        for the hourly or monthly data
+
+        Parameters
+        ----------
+        granularity: str, {'monthly', 'hourly'}, default: 'hourly'
+            The granularity of data being pulled
+
+        Returns
+        ----------
+        selection_dict: dict
+            A dictionary with all the time-related arguments of the
+            selection dict filled out
+        """
+        years = [str(year) for year in range(1979, 2019 + 1)]
+        months = ['{:02d}'.format(month) for month in range(1, 12 + 1)]
+        days = ['{:02d}'.format(day) for day in range(1, 31 + 1)]
+        times = ['{:02d}:00'.format(hour) for hour in range(24)]
+
+        selection_dict = {
+            'year': years,
+            'month': months,
+            'time': times,
+        }
+        if granularity == 'hourly':
+            selection_dict['day'] = days
+        return selection_dict
+
+    @staticmethod
+    def create_area(region: Region) -> str:
+        """Create an area string for the CDS API from a Region object
+
+        Parameters
+        ----------
+        region: Region
+            A Region to be exported from the CDS warehouse
+
+        Returns
+        ----------
+        area: str
+            A string representing the region which can be passed to the CDS API
+        """
+        x = [region.latmax, region.lonmin, region.latmin, region.lonmax]
+
+        return "/".join(["{:.3f}".format(x) for x in x])
+
+    @staticmethod
+    def _filename_from_selection_request(time_array: List, time: str) -> str:
+        if len(time_array) > 1:
+            if time in {'month', 'year'}:
+                warnings.warn(f'More than 1 {time} of data being exported! '
+                              'Export times may be significant.')
+            time_array.sort()
+            time_array_str = f'{time_array[0]}_{time_array[-1]}'
```

Comment:
@gabrieltseng another question - does it really make sense to have a separate folder for the two months?

```
.
├── README.md
└── raw
    ├── reanalysis-era5-single-levels
    │   └── precipitation
    │       ├── 1979_2019
    │       │   ├── 01_12
    │       │   └── 12
    │       ├── 1980
    │       │   ├── 09
    │       │   └── 12
    │       └── 1980_2010
    │           └── 09_12
```

How will the files be saved in folder `raw/reanalysis-era5-single-levels/precipitation/1980_2010/09_12`? 

Because currently they are written as: `f'{days}.nc'` but isn't that going to lead to a whole load of files overwriting each other because you've got every file from `1980 - 2010` and every month from `September - December` all saved by their `day` names ?